18329 Forms refactor based on Foundation for Apps

myabc opened the pull request:
Adresses:
- https://community.openproject.org/work_packages/18329
- https://community.openproject.org/work_packages/18454
- https://community.openproject.org/work_packages/18376
- https://community.openproject.org/work_packages/18785
- https://community.openproject.org/work_packages/18523

Continuation of #2294.
- [ ] Update, test TabularFormBuilder
- [x] Use TabularFormBuilder throughout application - @ulferts<br/> 
  - [x] project settings
    - [x] project main attributes
    - [x] project modules
    - [x] categories
    - [x] repositories
    - [x] boards
  - [x] versions
     (i.e. make default)
- [x] Refactor, cleanup TabularFormBuilder
-  ~~Display Rails form errors correctly~~<br/> 
     (adapt or remove `ActionView::Base.wrap_with_error_span` monkey patch)
- [x] Update helpers generating forms
  - [x] CustomFieldsHelper - @ulferts
  - [x] ProjectsHelper
  - [x] SettingsHelper - @myabc
- [x] Update views 
  - [x] Account (@myabc)
  - [x] Activities
  - [x] Auth/LDAP Sources
  - [x] Dependencies
  - [x] Enumerations
  - [x] Forum/Boards/Messages (@ulferts)
  - [x] Groups
  - [x] News
  - [x] Planning Type Colors
  - [x] Project Settings
  - [x] Project Types
  - [x] Projects
  - [x] Roles
  - [x] Settings (@myabc)
  - [x] Statuses
  - [x] Status Reportings
  - [x] Time Entries
  - [x] Timelines 
  - [x] Types
  - [x] Users
  - [x] Versions
  - [X] Wiki
  - [x] Work Packages @ulferts
    - [x] Bulk / Moves actions
  - [x] Workflows (@myabc)
  - [x] Time entry reports (work_packages/:id/time_entries/report) (@ulferts)
- [x] Update OP plugins
  - [x] Costs (https://github.com/finnlabs/openproject-costs/pull/121)
  - [x] Backlogs (https://github.com/finnlabs/openproject-backlogs/pull/120)
  - [x] Global roles (https://github.com/finnlabs/openproject-global_roles/pull/32)
  - [x] Meetings (https://github.com/finnlabs/openproject-meeting/pull/62)
  - [x] Documents (https://github.com/opf/openproject-documents/pull/26)
  - [x] PDF Export (https://github.com/finnlabs/openproject-pdf_export/pull/34)
  - ~~Checkout (https://community.openproject.org/work_packages/18685)~~ 
- [x] check and fix specs 

---

0xF013 committed:
Migrate to ui-select in Columns dialog

User Story # 18372

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Migrate to ui-select in Group by dialog

User Story # 18372

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add basic spec for TabularFormBuilder

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add basic spec for SettingsHelper

Signed-off-by: Alex Coles <alex@alexbcoles.com>

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://feature-18329-foundation-apps-framework-forms-refactor.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://feature-18329-foundation-apps-framework-forms-refactor.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

ulferts commented on spec/helpers/settings_helper_spec.rb:
```diff
@@ -0,0 +1,197 @@
+      helper.setting_multiselect :field, [['Popsickle', '1'], ['Jello', '2'], ['Ice Cream', '3']]
+    }
+
+    xit 'should have three labels' do
```
why is this disabled?

ulferts commented on spec/helpers/settings_helper_spec.rb:
```diff
@@ -0,0 +1,197 @@
+  include Capybara::RSpecMatchers
+
+  shared_examples_for 'labelled' do
+    it { is_expected.to have_selector 'label' }
```
couldn't we also check for the value of the label?

myabc replied:
@ulferts we should, yes.

ulferts commented on spec/lib/tabular_form_builder_spec.rb:
```diff
@@ -0,0 +1,304 @@
+  let(:builder)  { TabularFormBuilder.new(:user, resource, helper, {}, nil) }
+
+  shared_examples_for 'labelled' do
+    it { is_expected.to have_selector 'label' }
```
please see above

myabc commented on spec/helpers/settings_helper_spec.rb:
```diff
@@ -0,0 +1,197 @@
+require 'ostruct'
+
+describe SettingsHelper, type: :helper do
+  include Capybara::RSpecMatchers
```
@ulferts would like to get your feedback on using `Capybara::RSpecMatchers` here. `Capybara::RSpecMatchers` are often useful outside of integration specs (indeed we also use them in the API v3 request specs), but I'm concerned they might be the wrong level of abstraction here…

In integration specs we're typically interested in the data being displayed and not the exact structure of the DOM. In these Helper specs, we _are interested_ in the DOM structure. If Rails or our own application code happens to append additional attributes, that is also something we care about.

However, I find doing exact String matching for Helper output to be cumbersome (you need to ignore whitespace differences, for a start). Changes in attribute order (that have no effect on what is actually rendered in the browser) mean a lot of [extra spec maintenance](https://github.com/opf/openproject/commit/4b90766bc0c00d15c7c1841897c768189ade7a7f).

Alternatives are Rails' `assert_select`, but I don't really care for the nested blocks API.

WDYT?

ulferts replied:
@myabc I have no aversions against using Caypbara::RSpecMatchers here if we can avoid having to rewrite what they can provide us. 

A thin wrapper around it might make sense if we are to reuse dom creation testing. But that can be done later once we identify the need.

Regardless of the technical means used, as this is a unit test we should probably be more strict about the expected output. e.g. specify the container dom object ('p'/'li' with classname) we expect the form element to be in.

myabc replied:
@ulferts I figured out what I need/want for this type of testing: `json_spec`-like assertions, but for XML/HTML nodes.

myabc committed:
Update TabularFormBuilder to apply form classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Test more exact output in TabularFormBuilder spec

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Obey line length limits in TabularFormBuilder spec

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Test more exact output in SettingsHelper spec

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Consolidate form styles, reworking w/CSS flexboxes

This patch removes most existing form styles in favour of a new,
more uniform family of form classes under a `form` module.

User Story # 18329.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document text areas

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Show form layouts full width

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update SettingsHelper to apply form classes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented on spec/helpers/settings_helper_spec.rb:
```diff
@@ -0,0 +1,222 @@
+
+    it 'should output element' do
+      expect(output).to include %{
+        <input id="settings_field" name="settings[field]" type="text" value="important value" />
```
@ulferts in some views we're using these helper methods with the `TabularFormBuilder`, so the right classes are applied to each form element. In some views, we're using these helper methods with a standard `form_tag`. Do you have any idea of how I could easily enable/disable `TabularFormBuilder` within this spec?

ulferts replied:
I am not certain if I understand what you want to do and for what I understand, manipulating the contant (which would not be clean) is the only thing that comes to mind.

myabc replied:
@ulferts Ignore my previous comment. I was confused. `SettingsHelper` relies on `FormTagHelper` and not `FormHelper`, thus pays no attention to `ActionView::Helpers::FormBuilder`.

myabc committed:
Add, document form field extra actions styling

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, document support for multiple fields per row

This change necessitates an additional container element in the DOM
hierarchy.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update TabularFormBuilder for multiple fields/row

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Create container classes for all form field types

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Do not double wrap fields with no_label option

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add specs for TabularFormBuilder localized fields

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tweak alignment of fields w/multi. field container

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix field container width (forms w/o wide labels)

Broken in 2c3bccc1.

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Correct form class labelled_tabular_form_for uses

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Extract duplicated shared examples into own file

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update SettingsHelper for multi. fields/row change

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix SettingsHelper field label: false handling

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Make SettingsHelper apply classes to form elements

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix TabularFormBuilder handling of custom classes

Multiple classes were not delimited properly with a space.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented on lib/tabular_form_builder.rb:
```diff
@@ -70,24 +72,57 @@ def #{selector}(field, options = {})
 
     id = element_id(translation_form) if translation_form
-    label_options = { class: (@object && @object.errors[field] ? 'error' : nil) }
+    label_options = { class: (@object && @object.errors[field] ? '' : '') } # FIXME
```
@ulferts @floriank this needs checking

myabc committed:
Update custom fields form for new form styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
use tabular form builder for project module selection

* Extends the tabular form builder with the ability to generate check boxes for values that are an array.
* Fixes module_enabled? to actually return a boolean
* Adapts projects_controller to changed params structure.

myabc committed:
Add classes to TabularFormBuilder localized fields

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix TabularFormBuilder localized field support

Broken in refactoring 65de31a7.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/18329-foundation-apps-framework-forms-refactor

ulferts committed:
harmonize project module selection view

ulferts committed:
adapt project main settings to new form style

myabc committed:
Living Style Guide: restrict height of code blocks

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Set proper line height for labels, extra actions

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, document vertical forms layout/modifier

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply form styling to top menu login form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply form styling to login form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add OpenProject::FormTagHelper with styled helpers

Provides most of the helper methods provided by `ActionView::Helpers::
FormTagHelper`, but with relevant CSS classes, wrappers applied.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
add custom field to form builder - apply to version

Commit:
udpate wiki forms to use new form builder

ulferts committed:
extract custom_fields form logic into separate builder

ulferts committed:
refactor category views to new form layout

Commit:
use new markup for wiki

Commit:
Merge branch 'feature/18329/wiki_view' into feature/18329-foundation-apps-framework-forms-refactor

Commit:
use new markup in reportings forms

Commit:
fix label handling

ulferts committed:
refactor repository admin views to new form layout

ulferts committed:
refactor board admin views to new form layout

ulferts committed:
add missing separator

ulferts committed:
harmonize custom_value forms

ulferts committed:
remove unnecessary box around project form

myabc committed:
Update SettingsHelper to use styled helper methods

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
update view for news

Commit:
Merge branch 'feature/18329/news_view' into feature/18329-foundation-apps-framework-forms-refactor

myabc committed:
Add, document form field no-label modifier

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented on lib/tabular_form_builder.rb:
```diff
@@ -70,24 +54,107 @@ def #{selector}(field, options = {})
+    input_options = options.merge(multiple: true, no_label: true, checked: checked)
+    input = check_box(field, input_options, value, '')
+
+    label + container_wrap_field(input, 'check-box', options)
```
@ulferts unfortunately, it looks like checkboxes are being wrapped in a container twice here

![seeded_project_-_settings_-_openproject](https://cloud.githubusercontent.com/assets/755/6107432/26c04c98-b06b-11e4-9621-f8d8d0557fa9.png)

myabc commented on spec/support/shared/forms_html.rb:
```diff
@@ -0,0 +1,68 @@
+end
+
+shared_examples_for 'wrapped in container' do |container = 'field-container'|
+  it { is_expected.to have_selector "span.form--#{container}" }
```
@ulferts it's probably a good idea for us to change this line to 

``` ruby
it { is_expected.to have_selector "span.form--#{container}", count: 1 }
```

this would make it easier to catch issues like https://github.com/opf/openproject/pull/2492/files#r24328500

myabc commented on spec/lib/tabular_form_builder_spec.rb:
```diff
@@ -0,0 +1,443 @@
+        expect(output).to include %{
+          <input class="form--text-field"
+            id="user_translations_attributes_0_name"
+            name="user[translations_attributes][0][name]" no_label="true" size="30" type="text" />
```
anyone wants to tackle getting rid of the `no_label` attribute here?

Commit:
update project dependency forms

Commit:
Merge branch 'feature/18329/dependency_view' into feature/18329-foundation-apps-framework-forms-refactor

myabc committed:
Add, document styling for multi. checkbox fields

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update SettingsHelper for multi. checkbox styling

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling in administration settings views

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
[wip] Teach settings_text_field a unit option

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
rudimentary styling of the attachments form

ulferts committed:
style boards form

myabc committed:
Further fix SettingsHelper label: false handling

See 89572a28.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove SettingsHelper outer wrap w/label: false

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply vertical field styles to email notifications

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/18329-foundation-apps-framework-forms-refactor

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	app/views/messages/show.html.erb
	app/views/wiki/edit.html.erb
	app/views/wiki/new.html.erb

ulferts committed:
style work package form

Commit:
update view for general tab in timelines form

myabc committed:
Fix form separator class on login forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to register, lost password forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Use TabularFormBuilder for register custom fields

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to admin. auth sources forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to admin. enumerations forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to group forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to My forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to admin. roles forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to admin. statuses forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to admin. types forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to work packages bulk, move forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply styling to workflow copy form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
fix tests for wiki and permitted params

Commit:
fix project associations test

Commit:
fix partial issue in cuke for copy projects

myabc committed:
Apply styling to user forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
move integration test to scenario

ulferts committed:
refactor time entries

Commit:
fix features for work packages

The fix consists mostly of rewriting the test for a missing subject into
a test for a subject that is too long.

This is due to the fact that with the new form builder, the forms make
use of the `required` attribute for form fields in HTML5 documents. This
prevents the form to be sent w/ an empty, but required field.

myabc committed:
Remove (apparently) unused random_password partial

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
remove duplicate form-container + unnecessary dom attributes

ulferts committed:
fix work package helper specs

ulferts committed:
fix attachment feature spec

ulferts committed:
fix work package controller spec

ulferts committed:
removes expected unnecessary attribute from spec

myabc commented on app/helpers/work_packages_helper.rb:
```diff
@@ -557,15 +653,21 @@ def work_package_form_estimated_hours_attribute(form, work_package, _locals = {}
   def work_package_form_custom_values_attribute(_form, work_package, _locals = {})
-    work_package.custom_field_values.map do |value|
-      field = custom_field_tag_with_label :work_package, value
+    fields = work_package.custom_field_values.map do |value|
```
@ulferts what Hound says

myabc committed:
Remove default border, padding from forms

Instead provide, document bordered modifier.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply bordered form styling to login forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply bordered form styling to work package forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove field row class in favour of using grid

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts commented:
https://community.openproject.org/work_packages/18454

myabc committed:
Pad all but the last field element container

myabc committed:
Pad all but last form field inside a grid-block

Commit:
use new formbuilder for timelines form

myabc committed:
Remove apparent left, right padding on form fields

(padding is actually on the inner label, field container elements)

myabc committed:
Reduce form field bottom margin slightly

This value matches the margin applied to key-value attributes.

ulferts committed:
fix accessibility "lang" tags on custom fields

myabc committed:
Apply form section styling to user forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply form section styling to My forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, document styling for form sections

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, document styling for form fieldsets

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply form fieldset styling to user forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply section, fieldset styles to settings views

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply fieldset styling to attachments form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply section,fieldset styles to custom fields form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Replace <p>-tags with <div> in custom fields form

* Fix ERb case-statement indentation.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Apply section styles to register, lost password

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Arrange roles fieldsets in a grid

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
rework the opnening and closing of fieldsets

Commit:
fix hound complaints

myabc commented on app/assets/javascripts/timelines_select_boxes.js:
```diff
@@ -147,24 +147,22 @@ jQuery(document).ready(function($) {
+                            .not("[type='radio']")
+                            .not("[class^='select2-']")
+                            .not(".button")
+                            .not("[type='hidden']")
```
@floriank I configured JSHint with `asi: false`, but I'm not 100% sure this is the right choice… if you prefer to rely on ASI then let me know / @0xF013

floriank replied:
@myabc It's fine, it's fine, i am just not used to writing pure JavaScript anymore.

Commit:
fix even moar hound complaints

Commit:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of github.com:opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

Commit:
fix more hound complaints about JS

myabc committed:
[wip] Use styled helpers in CustomFieldsHelper

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
update lanning type colors form

myabc committed:
Apply fieldset styling to colors form

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
fix the attribute field for color

floriank commented:
woops. should be fixed @myabc

Commit:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of github.com:opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

Commit:
update the project types forms

Commit:
fix required on the version name field

ulferts committed:
adapt spec to changed dom

Commit:
fix missing confirm_destroy actions on some forms

myabc committed:
IE fix [wip]: hide form select arrows

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

ulferts committed:
fix test_unit tests

ulferts committed:
apply new style to additional wiki page forms

ulferts committed:
fix previewing work package journal notes

myabc committed:
IE fix: Use `flex-basis: auto` for vertical layout

We are not setting an explicit height on the flex container, so auto
actually makes more sense in this case.

Issue possibly relates to IE bug:
https://connect.microsoft.com/IE/feedback/details/802625/min-height-and-flexbox-flex-direction-column-dont-work-together-in-ie-10-11-preview

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/18329-foundation-apps-framework-forms-refactor

ulferts committed:
remove harmfull lang option

ulferts committed:
fix parameter naming

Commit:
fix the color indicator in the colors form

Commit:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of github.com:opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

Commit:
remove logging

Commit:
fix copy watchers feature

Commit:
fix the features for work packages

fix switch type feature for work packages

fix the features for preview on work packages

fix the feature for localized log times

fix error on update feature

fix editable fields

fix the bulk delete action w/ feature

fix udpate feature

ulferts committed:
adapt cuke select step to changed dom for select2

Commit:
fix wiki features

this is fixed by extending the collection_checkbox method of the
TabularFormBuilder by a prefix-option. This is so that the labels are
properly set for the inidividual check boxes

ulferts committed:
adapts association form styles

Commit:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of github.com:opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

ulferts committed:
rework fixing collection_check_box labels

myabc committed:
Remove modal dialog form styling overrides

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update Save dialog to use new form styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update Settings dialog to use new form styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix missing label translation in Save dialog

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
fix some reporter features (WIP)

Commit:
fix timelines form

Commit:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of github.com:opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

Commit:
fix reportings feature

Commit:
ause correct field for modal window testing

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
fix label for rename query modal

Signed-off-by: Florian Kraft <f.kraft@finn.de>

myabc committed:
Remove max-widths from vertical form boxes

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/18329-foundation-apps-framework-forms-refactor

myabc committed:
Fix login provider item width for border-box

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document login auth providers

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Commit:
use tabindex for login

Signed-off-by: Florian Kraft <f.kraft@finn.de>

myabc committed:
Specify a width for top menu login popup

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Adjust login form width for border-box

Signed-off-by: Alex Coles <alex@alexbcoles.com>

floriank commented:
I'll start reviewing this.

Commit:
use new markup in workflow edit form

Signed-off-by: Florian Kraft <f.kraft@finn.de>

myabc committed:
Merge branch 'dev' into feature/18329-foundation-apps-framework-forms-refactor

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	app/assets/stylesheets/content/_work_packages.sass

myabc commented:
@ulferts @floriank 
- There are new mockups for error styling. I'd like to handle that as a separate work package.
- Updating the meeting plugin would be straight-forward, if it weren't for the  `MessageAccessibilityHelper#accessible_time_select` helper. I'd love to get rid of this piece of untested code! 
  To obtain the desired appearance, I want to wrap select boxes individually in `form--select-container` and that's not something this helper, or the Rails `time_select` helper for that matter can do. 
  
  Options are:
  - CSS hack – or at least a special case. 
    
    _Advantage:_ avoid touching `MessageAccessibilityHelper`.
    _Disadvantage:_ Form code are already relatively complicated, and I'd like to keep this new Sass clean.
  - Monkey-patching `time_select` or `accessible_time_select`. 
    
    _Advantage:_ avoid dirtying our stylesheets. 
    _Disadvantage:_ we'd really want to add test coverage first. We should probably also move these methods into core.
  - Removing `time_select` altogether, replacing it with an [HTML 5 input="time" element](http://www.wufoo.com/html5/types/4-date.html) along with some sort of fallback.
    
    _Advantage:_ we'll use this code elsewhere. Future-proof.
    _Disadvantage:_ not all browsers supported. probably needs some JavaScript polyfill.

floriank commented:
I'd probably be most content with option nr. 3 - but the browsers that support such inputs are rather rare (http://caniuse.com/#feat=input-datetime), so we'd definitely need a fallback for that.

0xF013 commented:
IE strikes again

floriank commented:
As does FF and Safari :dart:

Commit:
fix markup for static label in work package view

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of github.com:opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

Commit:
please the hound

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
use form builder for boards component

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
aply new form builders to time entry reports

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
introduce form--grouping and apply to timelines

ulferts committed:
add html_safe on no_label

ulferts commented:
https://community.openproject.org/work_packages/18376

ulferts committed:
fix calendar styling

ulferts commented:
https://community.openproject.org/work_packages/18785

0xF013 committed:
allow empty value for group by

myabc committed:
Merge pull request #2612 from opf/feature/18329-foundation-apps-framework-forms-refactor-timelines

Feature/18329 foundation apps framework forms refactor timelines

ulferts committed:
prevent scrollbars on scm diff views

ulferts committed:
style work package by category on version show

ulferts commented:
https://community.openproject.org/work_packages/18523

ulferts committed:
fix cuke by restyling timelog

ulferts committed:
reduce deprecation warnings by adding translations

0xF013 committed:
sorting dialog

Commit:
fix functional messages controller test

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
fix timelog controller test

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
fix timelog controller spec

Signed-off-by: Florian Kraft <f.kraft@finn.de>

myabc committed:
Merge pull request #2549 from opf/feature/18372-migrate-to-ui-select-in-modals

18372 migrate to ui select in modals

myabc committed:
Add styling for tables within forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

floriank commented:
@0xF013 @myabc I'll do the cukes.

myabc commented:
@floriank thanks! @0xF013 will take a look at Karma.

Commit:
fix features for work packages

this should fix the failing test introduced by the ui-select PR

Signed-off-by: Florian Kraft <f.kraft@finn.de>

0xF013 committed:
removed the now useless tests of sorting options select2

0xF013 committed:
Merge branch 'feature/18329-foundation-apps-framework-forms-refactor' of https://github.com/opf/openproject into feature/18329-foundation-apps-framework-forms-refactor

0xF013 commented:
@myabc we can wrap each ui.select + radios in a normal div. Should work

myabc committed:
Peg foundation-apps at 1.0.2

In 1.0.3 a regression was introduced in Foundation grid, whereby
`.grid-content` nested inside a `.grid-block` will no longer take up
an equal amount of space.

Commit in question:
https://github.com/zurb/foundation-apps/commit/fded6510da6acbd97e608f580527a08bfc5c2823

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix work package feature specs for ui-select

See d1a3b20c.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Port Firefox select fixes from Foundation 1.0.3

See https://github.com/zurb/foundation-apps/commit/43da85985770e430ce45a8c10fdd6bed29cf5096

(Upgrading to 1.0.3 is not yet possible. See e22c48a6)

Bug # 18388

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
https://community.openproject.org/work_packages/18388

myabc commented:
@floriank @ulferts There's obviously still more I'd like to do, but provided Travis agrees, this should now be mergeable. Issues discussed early today should have been dealt with.

myabc committed:
Bump Angular to 1.3.14

Signed-off-by: Alex Coles <alex@alexbcoles.com>
